sdks/java/core/src/main/java/org/apache/beam/sdk/util/NullCredential.java

```diff
@@ -0,0 +1,53 @@
+/*
```

Comment:
For cases when the user doesn't have credentials and is trying to run a pipeline, how does there error message now change for an unauthorized user. Before we would fail on creation of the client, but now we might try to stage files or contact BQ to see if datasets exist. Is the message as to why it failed same, better or worse off for various GCP service use cases.